### DISCUSSION TITLE
`cargoNextest` `checkPhaseCargoCommand` as `sudo`?

### DISCUSSION THREAD

**Discussion by eureka-cpu** [ipetkov/crane#681] [General]
Hello, I have some things in a few cargo tests that require sudo privilages and it would be nice to just add `sudo` to the `checkPhaseCargoCommand` invokation. Is something like this possible? Am I barking up the wrong tree?

The function in the test case is `create_dir_all` which fails with `Insufficient permissions`.

**Comment by eureka-cpu** [comment]
Running `nix flake check` or `sudo nix flake check`, the test fails. However, doing `sudo nix develop` then `cargo nextest r`, and the test passes. I figure if there's a way to just use `sudo` for the `cargoNextest` check phase it would solve the problem. Also open to other suggestions!

**Reply by dpc** [reply]
Is it possible to even run `sudo` in Nix derivation? I doubt it, but maybe I'm wrong.

**Reply by eureka-cpu** [reply]
Another thing I've considered is adding more relaxed permissions for what the test requires to pass, but this would mean everyone on my team would need to do the same which isn't ideal. I may end up just making those tests have an extra cargo feature, and only run in CI.

**Reply by dpc** [reply]
Life is tough sometimes, but generally tests that require `sudo` are anything but isolated, and thus not a great tests to have. I would consider using https://wiki.nixos.org/wiki/NixOS_VM_tests to run such tests in a VM. It's actually quite low overhead.

**Reply by ipetkov** [reply]
I second that NixOS VM tests (or changing how the tests are done) is the way to go. Even if you could change the builder scripts to call `sudo`, I don't think that will work since the build is sandboxed and run as a `nixbuild` user which certainly shouldn't have sudo permissions (if it did, then any build could become root and tamper with the store if it wanted to)
